### PR TITLE
Fix compile warnnings on gcc 8

### PR DESCRIPTION
**Pull request by cathay4t**
Tested on Fedora rawhide `gcc-8.0.1-0.19.fc29.x86_64`

### CODE REVIEW

File: c_binding/libsg.c [hunk 1]
```diff
@@ -485,14 +486,14 @@ int _sg_io_vpd(char *err_msg, int fd, uint8_t page_code, uint8_t *data)
...
-    char sense_err_msg[_LSM_ERR_MSG_LEN];
+    char sense_err_msg[_LSM_ERR_MSG_LEN / 2];
     ssize_t data_len = 0;
 
     assert(err_msg != NULL);
     assert(fd >= 0);
     assert(data != NULL);
 
-    memset(sense_err_msg, 0, _LSM_ERR_MSG_LEN);
+    memset(sense_err_msg, 0, sizeof(sense_err_msg)/sizeof(char));
```
Comment (tasleson) [inline c_binding/libsg.c]: `memset` clears a number of bytes, the statement `sizeof(sense_err_msg)/sizeof(char)` gives you the count or number of items stored in the memory which is not what what we want.  In this case we happen to be OK as `sizeof(char)` happens to be 1, thus N/1 == N.  Please simply change all the `memset` calls to use `sizeof(thing)`.

**Comment by tasleson** [comment]
@cathay4t Have you compiled this on RHEL6/7?

File: c_binding/libsg.c (hunk 1 above)
Comment (cathay4t) [reply c_binding/libsg.c]: I will fix it.

**Comment by cathay4t** [comment]
The travis CI will check RHEL/Centos 7 using `make distcheck`.
Let me see whether I could add CI for RHEL/Centos 6.

**Comment by cathay4t** [comment]
http://people.redhat.com/fge/lsm/libstoragemgmt-1.6.1-2.el7.src.rpm

**Comment by cathay4t** [comment]
RHEL 6.9 'make distcheck' pass.

**Comment by cathay4t** [comment]
New patch uploaded to enable travis CI for Centos 6 `make check`.

**Comment by cathay4t** [comment]
GCC 8.1 released. PR tested against it.
Merged.

**Pull request merged**